pkg/cli/app_spec.go

```diff
@@ -16,9 +16,10 @@ type AppSpec struct {
 	Prompt      Prompt
 	RPrompt     Prompt
 
-	OverlayHandler Handler
-	Abbreviations  func(f func(abbr, full string))
-	QuotePaste     func() bool
+	OverlayHandler    Handler
+	Abbreviations     func(f func(abbr, full string) bool)
+	WordAbbreviations func(f func(abbr, full string) bool)
```

Comment:
Call this SmallWordAbbreviations; same for the edit API: `edit:small-word-abbr`.

For symmetry, there should also be `edit:word-abbr` and `edit:alnum-word-abbr`.

Reply:
I'll include "small" in the respective tokens and adjust the documentation and implementation accordingly. In particular with respect to the idea of supporting arbitrary "small word" transitions per your other comment. It's not clear that supporting "alnum" abbreviations, as I've implemented in the PR up to this point, is justified given the more general "small word" abbreviation mechanism.